Portworx Volume Plugin

adityadani opened the pull request:
**What this PR does / why we need it**:
This PR adds a new volume plugin in Kubernetes for Portworx Volumes. Portworx pools available block storage on nodes and carves out volumes from it. More info about Portworx can be found [here](http://docs.portworx.com) The volume plugin supports
- PVs
- PVCs
- Dynamic Provisioning using Storage Classes

**Special notes for your reviewer**:
I have divided the change into 4 commits
1. Code for the Volume Plugin
2. Configuration Files, Examples, Validation Files
3. Godeps
4. Generated code and Documentation

```release-note
Portworx Volume Plugin added enabling [Portworx](http://www.portworx.com) to be used as a storage provider for Kubernetes clusters. Portworx pools your servers capacity and turns your servers or cloud instances into converged, highly available compute and storage nodes.
```

adityadani committed:
Portworx Volume Driver in Kubernetes

- Add a new type PortworxVolumeSource
- Implement the kubernetes volume plugin for Portworx Volumes under pkg/volume/portworx
- The Portworx Volume Driver uses the libopenstorage/openstorage specifications and apis for volume operations.

Changes for k8s configuration and examples for portworx volumes.

- Add PortworxVolume hooks in kubectl, kube-controller-manager and validation.
- Add a README for PortworxVolume usage as PVs, PVCs and StorageClass.
- Add example spec files

Handle code review comments.

- Modified READMEs to incorporate to suggestions.
- Add a test for ReadWriteMany access mode.
- Use util.UnmountPath in TearDown.
- Add ReadOnly flag to PortworxVolumeSource
- Use hostname:port instead of unix sockets
- Delete the mount dir in TearDown.
- Fix link issue in persistentvolumes README
- In unit test check for mountpath after Setup is done.
- Add PVC Claim Name as a Portworx Volume Label

Generated code and documentation.
- Updated swagger spec
- Updated api-reference docs
- Updated generated code under pkg/api/v1

Godeps update for Portworx Volume Driver
- Adds github.com/libopenstorage/openstorage
- Adds go.pedge.io/pb/go/google/protobuf
- Updates Godep Licenses

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/39535)
<!-- Reviewable:end -->

rootfs commented on pkg/api/types.go:
```diff
@@ -950,6 +956,17 @@ type PhotonPersistentDiskVolumeSource struct {
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs". Implicitly inferred to be "ext4" if unspecified.
+	// +optional
+	FSType string
```
do you need `readonly` flag?

rootfs commented on pkg/volume/portworx/doc.go:
```diff
@@ -0,0 +1,19 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
happy 2017

rootfs commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,388 @@
+func getVolumeSource(
+	spec *volume.Spec) (*v1.PortworxVolumeSource, bool, error) {
+	if spec.Volume != nil && spec.Volume.PortworxVolume != nil {
+		return spec.Volume.PortworxVolume, spec.ReadOnly, nil
```
`spec.ReadOnly` is not trusted in this case. You need to provide readonly in portworx volume source.

adityadani replied:
Ah! I will add a Readonly flag in PortworxVolumeSource.

rootfs commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,388 @@
+	if err := b.manager.MountVolume(b, dir); err != nil {
+		return err
+	}
+
```
call `SetVolumeOwnership` to chown gid

rootfs commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,388 @@
+
+// SetUpAt attaches the disk and bind mounts to the volume path.
+func (b *portworxVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
+
```
removfe blank line

rootfs commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,388 @@
+// Unmounts the bind mount, and detaches the disk only if the PD
+// resource was the last reference to that disk on the kubelet.
+func (c *portworxVolumeUnmounter) TearDownAt(dir string) error {
+	glog.V(4).Infof("Portworx Volume TearDown of %s", dir)
```
use `util.UnmountPath`  in `pkg/volume/util`

rootfs commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,388 @@
+		return err
+	}
+
+	if err := c.manager.DetachVolume(c); err != nil {
```
double check if `dir` is till a mount point before detaching and removing it.

rootfs commented on pkg/volume/portworx/portworx_test.go:
```diff
@@ -0,0 +1,221 @@
+	if !contains(plug.GetAccessModes(), v1.ReadWriteOnce) {
+		t.Errorf("Expected to support AccessModeTypes:  %s", v1.ReadWriteOnce)
+	}
+	if contains(plug.GetAccessModes(), v1.ReadOnlyMany) {
```
add a test case for readwritemany

adityadani replied:
Will do.

rootfs commented on pkg/volume/portworx/portworx_test.go:
```diff
@@ -0,0 +1,221 @@
+		t.Errorf("Provision() failed: %v", err)
+	}
+
+	if persistentSpec.Spec.PersistentVolumeSource.PortworxVolume.VolumeID != "portworx-test-vol" {
```
nit: make `portworx-test-vol` a const

rootfs commented on pkg/volume/portworx/portworx_test.go:
```diff
@@ -0,0 +1,221 @@
+	}
+	cap := persistentSpec.Spec.Capacity[v1.ResourceStorage]
+	size := cap.Value()
+	if size != 1024*1024*1024*100 {
```
for consistency, 1024*1024*1024*100=100G != 100Mi (yes, it is a problem in gce-pd test too)

adityadani replied:
@rootfs I am not sure about the consistency part. With what should this be consistent ?

rootfs commented on pkg/volume/portworx/portworx_util.go:
```diff
@@ -0,0 +1,168 @@
+	requestGB := int(volume.RoundUpSize(capacity.Value(), 1024*1024*1024))
+
+	var labels map[string]string
+	if p.options.CloudTags == nil {
```
where does `CloudTags` come from?

adityadani replied:
Remnant of some old code. Previously I was trying to use CloudTags before Storage Classes. I will remove this section.

rootfs commented on pkg/volume/portworx/portworx_util.go:
```diff
@@ -0,0 +1,168 @@
+
+func (util *PortworxVolumeUtil) osdClient(hostName string) (osdvolume.VolumeDriver, error) {
+	if util.portworxClient == nil {
+		driverClient, err := volumeclient.NewDriverClient("", pxdDriverName, osdDriverVersion)
```
what is the empty parameter for?

adityadani replied:
The client can talk on a port or a unix socket. The first argument is for the hostname. We are initialzing the client to talk only over a unix socket, hence it is empty. I will add a constant for no hostname instead of empty string.

rootfs commented on pkg/volume/portworx/portworx_util.go:
```diff
@@ -0,0 +1,168 @@
+	}
+
+	specHandler := osdspec.NewSpecHandler()
+	spec, err := specHandler.SpecFromOpts(p.options.Parameters)
```
do you need to validate the parameters before using them?

adityadani replied:
The SpecHandler instance will validate all params and return an error if any. The parameters are validated w.r.t Portworx's Volume Spec.

rootfs commented on examples/experimental/persistent-volume-provisioning/README.md:
```diff
@@ -63,6 +63,30 @@ parameters:
+*  `repl`: replication factor [1..3] (default: `1`)
+*  `io_priority`: IO Priority: [high|medium|low] (default: `low`)
+*  `snap_interval`: snapshot interval in minutes, 0 disables snaps (default: `0`)
+*  `aggregation_level`: specifies the number of parts the volume can be aggregated from (default: `0`)
```
it is unclear from the context what aggregation means

rootfs commented on examples/volumes/portworx/README.md:
```diff
@@ -0,0 +1,370 @@
+
+## Examples
+
+The following examples assumes that you already have a running Kubernetes cluster with Portworx installed on all nodes.
```
a link to installation instruction?

rootfs commented on examples/volumes/portworx/README.md:
```diff
@@ -0,0 +1,370 @@
+        persistentVolumeReclaimPolicy: Retain
+        portworxVolume:
+          volumeID: "<vol-id>"
+	  fsType:   "<fs-type>"
```
ident

rootfs commented on examples/volumes/portworx/README.md:
```diff
@@ -0,0 +1,370 @@
+     Message:
+     Source:
+         Type:	      PortworxVolume (a Portworx Persistent Volume resource)
+	 VolumeID:   374093969022973811
```
indent

rootfs commented on pkg/kubectl/describe.go:
```diff
@@ -676,6 +678,12 @@ func printQuobyteVolumeSource(quobyte *api.QuobyteVolumeSource, w *PrefixWriter)
 }
 
+func printPortworxVolumeSource(pwxVolume *api.PortworxVolumeSource, w *PrefixWriter) {
+	w.Write(LEVEL_2, "Type:\tPortworxVolume (a Portworx Persistent Volume resource)\n"+
```
remove persistent (consistent with api description)

adityadani commented:
@rootfs I have addressed your comments and made a few modifications. Can you take a look?

kerneltime commented on examples/persistent-volume-provisioning/README.md:
```diff
@@ -63,6 +63,30 @@ parameters:
+*  `io_priority`: IO Priority: [high|medium|low] (default: `low`)
+*  `snap_interval`: snapshot interval in minutes, 0 disables snaps (default: `0`)
+*  `aggregation_level`: specifies the number of chunks the volume would be distributed into, 0 indicates a non-aggregated volume (default: `0`)
+*  `ephemeral`: ephemeral storage [true|false] (default `false`)
```
Markdown on GitHub treats it as a link..

rootfs commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,390 @@
+		// Mountpoint still exists
+		return fmt.Errorf("Failed to unmount Portworx Volume at %s", dir)
+	}
+
```
also remove `dir` here

adityadani replied:
The util.UnmountPath() routine would delete the dir if it is not a mountpoint.
If I do os.Remove here then it is going to error out with no such file or directory
Am I understanding this correctly?

rootfs replied:
what this additional check for?
```go
// Double check if the volume is unmounted
notMnt, _ := c.mounter.IsLikelyNotMountPoint(dir)
```

adityadani replied:
yes that is also been done in util.UnmountPath().
Will remove this too.
So it will look like this

1.util.UnmountPath()
2. call portworx.UnmountVolume
3. call portworx.DetachVolume

rootfs commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,390 @@
+	if err := b.manager.MountVolume(b, dir); err != nil {
+		return err
+	}
+	volume.SetVolumeOwnership(b, fsGroup)
```
if !readonly

rootfs commented:
@adityadani a few more comments, almost there

adityadani commented:
@rootfs There were a couple of conflicts generated over a period of time. I have rebased my changes. Can you please take a look?

rootfs commented:
@adityadani can you squash into 2 commits: one for godeps and vendor,  the other for portworx implementation? I'll lgtm after that.

rootfs commented:
@k8s-bot test this

adityadani commented:
@k8s-bot test this

adityadani commented:
@rootfs I have squashed all my changes into 2 commits. The previous jenkins runs failed in Bazel Build. I added a fix for that. Can you restart the runs?

rootfs commented:
@k8s-bot test this

rootfs commented:
@k8s-bot ok to test

rootfs commented:
@adityadani verify-bazel.sh failed with the following:

```
I0207 17:49:37.558] diffing /go/src/k8s.io/kubernetes/docs/api-reference against freshly generated docs
I0207 17:49:37.591] /go/src/k8s.io/kubernetes/docs/api-reference up to date.
I0207 17:49:37.597] [0;32mSUCCESS[0m  hack/make-rules/../../hack/verify-api-reference-docs.sh	109s
I0207 17:49:37.597] Verifying hack/make-rules/../../hack/verify-bazel.sh
I0207 17:49:42.007] 47,49d46
I0207 17:49:42.007] <         "//kubernetes/cluster/addons:all-srcs",
I0207 17:49:42.007] <         "//kubernetes/examples:all-srcs",
I0207 17:49:42.007] <         "//kubernetes/federation/cluster:all-srcs",
W0207 17:49:42.108] DRY-RUN: wrote "BUILD.bazel"
W0207 17:49:42.556] 
W0207 17:49:42.557] 1 BUILD files not up-to-date.
I0207 17:49:42.657] 
```

adityadani commented:
@rootfs The latest verify-bazel.sh succeeded.
However AWS e2e test failed with this error

```
W0208 07:50:47.286] 2017/02/08 07:50:47 e2e.go:948: Step '/workspace/kops delete cluster aws-kops-agent-pr-5-0.test-aws.k8s.io --yes' finished in 3.959688894s
W0208 07:50:47.287] 2017/02/08 07:50:47 e2e.go:186: Saved XML output to /workspace/_artifacts/junit_runner.xml.
W0208 07:50:47.288] 2017/02/08 07:50:47 e2e.go:230: Something went wrong: starting e2e cluster: kops configuration failed: exit status 1
```

rootfs commented:
/lgtm

adityadani commented:
The GCE and AWS e2e failed. I am not sure if this is an intermittent error and just requires a restart of the run

```
Expected error:
    <*errors.errorString | 0xc4203327a0>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
not to have occurred
```
@rootfs Any ideas what I can do?

rootfs commented:
@k8s-bot cvm gce e2e test this

adityadani commented:
@rootfs Thanks for enabling GCE e2e tests. Can you do the same for kops AWS e2e ?

rootfs commented:
@k8s-bot kops aws e2e test this

adityadani commented:
@rootfs Still the same error with AWS e2e. Is this an intermittent issue?

rootfs commented:
@k8s-bot kops aws e2e test this

adityadani commented:
@rootfs I had to rebase my changes. Can you start all the tests?

rootfs commented:
@k8s-bot test this

adityadani commented:
@rootfs All tests passed.
But I had to rebase it again

rootfs commented:
@adityadani i'll put lgtm back after your rebase

adityadani commented:
@rootfs I have rebase my changes

saad-ali commented:
@msau42 Can you take a first pass reviewing this new volume plugin. Once you're done I'll take a look. Let's make sure this makes code freeze for 1.6.

adityadani commented:
@saad-ali @msau42 Can you please take a look? I hope to check it in before Monday 20th (code freeze). I have a pending rebase, but I plan to address it along with any other comments from you guys.

msau42 commented:
/lgtm

rootfs commented:
@adityadani ping us back to add lgtm after rebase.

adityadani commented:
@rootfs @msau42 I have rebased my changes. Can you add the lgtm back ?

msau42 commented:
/lgtm

adityadani commented:
@rootfs @msau42 Thanks for adding the label.
I had a local file that caused my bazel build to succeed locally, but fail here. I have fixed it. Can you add it back? Sorry for the trouble.

rootfs commented:
/lgtm

rootfs commented:
@k8s-bot test this

adityadani commented:
@rootfs  Thank you for running the tests. Can you re-run GKE smoke e2e ?

msau42 commented:
@k8s-bot cvm gke e2e test this

adityadani commented:
@saad-ali @rootfs @msau42 All the tests have passed! Can someone add the approve label?

rootfs commented:
/approve

adityadani commented:
@rootfs @msau42 @saad-ali I had to rebase my code with the latest changes.
Can you take a look and enable the tests as well ?

rootfs commented:
/lgtm

saad-ali commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,386 @@
+		return nil
+	}
+
+	if _, err := b.manager.AttachVolume(b); err != nil {
```
Why not implement the attacher interface?

adityadani replied:
The attach operation for Portworx volumes is instantaneous. Hence I did not find it necessary to implement the Attacher interface especially the WaitForAttach. The following would simply send a REST call and attach the volume.

saad-ali replied:
The benefit of implementing the attacher interface is that the attach/detach operation are handled by the attach/detach controller instead of kubelet. Two benefits of this are 1) credentials required for attach/detach do not need to be available on node machines (for kubelet), and 2) if a node/kubelet becomes inaccessible, the master attach/detach controller can still ensure that volumes are detached (volumes are not stranded). For these reasons, you may want to consider refactoring in the future to implement the attacher interface. `WaitForAttach()` can always be a no-op if it is irrelevant.

saad-ali commented on pkg/volume/portworx/portworx.go:
```diff
@@ -0,0 +1,386 @@
+func (c *portworxVolumeUnmounter) TearDownAt(dir string) error {
+	glog.V(4).Infof("Portworx Volume TearDown of %s", dir)
+	// Unmount the bind mount inside the pod
+	util.UnmountPath(dir, c.mounter)
```
If there is an error here, it should be returned so that the caller can retry.

adityadani replied:
Sure! I will do the change.

saad-ali commented:
Looks good. Please squash your commits and update your original comment on this PR with a proper release-note, and I will LGTM/approve

adityadani commented:
@saad-ali I have rebased my changes and handled the error condition you mentioned. Can you take another look?

adityadani commented:
@k8s-bot test this

adityadani commented:
@saad-ali I had to rebase my changes. Can you take a look and also enable the tests? Also how can I get access to run tests through k8s-bot?

saad-ali commented:
@k8s-bot ok to test

saad-ali commented:
/lgtm
/approve

adityadani commented:
@saad-ali Previous bazel build failed and I doubt it was because I had one more new conflicting file. I have rebased my changes. Can you add the lgtm and approve labels back?

saad-ali commented:
This PR and the ScaleIO PR (https://github.com/kubernetes/kubernetes/pull/38924) and the mount options PR (https://github.com/kubernetes/kubernetes/pull/41906) will race: which ever ones get in first will force the others to to rebase.

saad-ali commented:
I tweaked the release note slightly and fixed the formatting.

saad-ali commented:
/lgtm

saad-ali commented:
```
=== RUN   TestStartingResourceVersion
2017-02-27 13:55:12.058013 I | integration: launching node6354447781065865164 (unix://localhost:node6354447781065865164.sock.bridge)
2017-02-27 13:55:12.063651 I | etcdserver: name = node6354447781065865164
2017-02-27 13:55:12.063738 I | etcdserver: data dir = /tmp/etcd979840010
2017-02-27 13:55:12.063805 I | etcdserver: member dir = /tmp/etcd979840010/member
2017-02-27 13:55:12.063882 I | etcdserver: heartbeat = 10ms
2017-02-27 13:55:12.063937 I | etcdserver: election = 100ms
2017-02-27 13:55:12.063998 I | etcdserver: snapshot count = 0
2017-02-27 13:55:12.064065 I | etcdserver: advertise client URLs = unix://127.0.0.1:21014.29798.sock
2017-02-27 13:55:12.064137 I | etcdserver: initial advertise peer URLs = unix://127.0.0.1:21013.29798.sock
2017-02-27 13:55:12.064214 I | etcdserver: initial cluster = node6354447781065865164=unix://127.0.0.1:21013.29798.sock
2017-02-27 13:55:12.072267 I | etcdserver: starting member 466649ce8409c65c in cluster 85fdcf7cb77e5fdb
2017-02-27 13:55:12.072466 I | raft: 466649ce8409c65c became follower at term 0
2017-02-27 13:55:12.072582 I | raft: newRaft 466649ce8409c65c [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2017-02-27 13:55:12.072657 I | raft: 466649ce8409c65c became follower at term 1
2017-02-27 13:55:12.085739 I | etcdserver: set snapshot count to default 10000
2017-02-27 13:55:12.085876 I | etcdserver: starting server... [version: 3.0.17, cluster version: to_be_decided]
2017-02-27 13:55:12.087573 I | membership: added member 466649ce8409c65c [unix://127.0.0.1:21013.29798.sock] to cluster 85fdcf7cb77e5fdb
2017-02-27 13:55:12.090829 I | integration: launched node6354447781065865164 (unix://localhost:node6354447781065865164.sock.bridge)
2017-02-27 13:55:12.163459 I | raft: 466649ce8409c65c is starting a new election at term 1
2017-02-27 13:55:12.163619 I | raft: 466649ce8409c65c became candidate at term 2
2017-02-27 13:55:12.163692 I | raft: 466649ce8409c65c received vote from 466649ce8409c65c at term 2
2017-02-27 13:55:12.163804 I | raft: 466649ce8409c65c became leader at term 2
2017-02-27 13:55:12.163896 I | raft: raft.node: 466649ce8409c65c elected leader 466649ce8409c65c at term 2
2017-02-27 13:55:12.172185 I | etcdserver: published {Name:node6354447781065865164 ClientURLs:[unix://127.0.0.1:21014.29798.sock]} to cluster 85fdcf7cb77e5fdb
2017-02-27 13:55:12.172375 I | etcdserver: setting up the initial cluster version to 3.0
2017-02-27 13:55:12.173368 N | membership: set the initial cluster version to 3.0
2017-02-27 13:55:12.289360 I | integration: terminating node6354447781065865164 (unix://localhost:node6354447781065865164.sock.bridge)
2017-02-27 13:55:12.297755 I | integration: terminated node6354447781065865164 (unix://localhost:node6354447781065865164.sock.bridge)
--- FAIL: TestStartingResourceVersion (0.24s)
panic: interface conversion: interface is nil, not *example.Pod [recovered]
	panic: interface conversion: interface is nil, not *example.Pod

goroutine 683 [running]:
panic(0x1794fe0, 0xc420205240)
	/usr/local/go/src/runtime/panic.go:500 +0x1ae
testing.tRunner.func1(0xc420bb4840)
	/usr/local/go/src/testing/testing.go:579 +0x474
panic(0x1794fe0, 0xc420205240)
	/usr/local/go/src/runtime/panic.go:458 +0x271
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/tests.TestStartingResourceVersion(0xc420bb4840)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/tests/cacher_test.go:525 +0xaf8
testing.tRunner(0xc420bb4840, 0x1a67500)
	/usr/local/go/src/testing/testing.go:610 +0xca
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x530
FAIL	k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/tests	6.222s
```

Not sure if that's a flake or related to your PR

adityadani commented:
Looks like a flake and also this test is mostly not related to my changes. I re-ran the tests on my dev box and it passed.

saad-ali commented:
@k8s-bot unit test this

saad-ali commented:
Verification is failing:

```
I0227 14:29:44.538]   diff '--ignore-matching-lines=^\s*\"Comment\"' -NauprB -I 'GoVersion.*\|GodepVersion.*' /go/src/k8s.io/kubernetes/staging/src/k8s.io/client-go/pkg/apis/extensions/types.go /go/src/k8s.io/kubernetes/staging/src/k8s.io/_tmp/pkg/apis/extensions/types.go
I0227 14:29:44.539]   --- /go/src/k8s.io/kubernetes/staging/src/k8s.io/client-go/pkg/apis/extensions/types.go	2017-02-27 13:57:20.060640923 -0800
I0227 14:29:44.539]   +++ /go/src/k8s.io/kubernetes/staging/src/k8s.io/_tmp/pkg/apis/extensions/types.go	2017-02-27 14:29:33.490824440 -0800
I0227 14:29:44.539]   @@ -917,6 +917,7 @@ var (
I0227 14:29:44.539]    	AzureDisk             FSType = "azureDisk"
I0227 14:29:44.539]    	PhotonPersistentDisk  FSType = "photonPersistentDisk"
I0227 14:29:44.539]    	Projected             FSType = "projected"
I0227 14:29:44.539]   +	PortworxVolume        FSType = "portworxVolume"
I0227 14:29:44.539]    	All                   FSType = "*"
I0227 14:29:44.539]    )
I0227 14:29:44.539]    
I0227 14:29:44.540]   /go/src/k8s.io/kubernetes/staging/src/k8s.io/client-go is out of date. Please run hack/update-staging-client-go.sh
I0227 14:29:44.540] FAILED   hack/make-rules/../../hack/verify-staging-client-go.sh	140s
```

adityadani commented:
./hack/update-all does not run "hack/update-staging-client-go.sh". I have fixed my changes and verified that "hack/update-staging-client-go.sh" succeeds.

saad-ali commented:
/lgtm

adityadani commented:
@saad-ali I had to rebase my changes with master. Can you add lgtm back and re-enable the tests?

saad-ali commented:
/lgtm

saad-ali commented:
@k8s-bot test this

adityadani commented:
@saad-ali non-CRI GCE e2e test failed. Not sure why, probably a flake?
Can you re-run it and also add the lgtm label?

msau42 commented:
@k8s-bot non-cri e2e test this

adityadani commented:
@msau42 Can you also add the lgtm label?

msau42 commented:
/lgtm
